<launch>
  <arg name="rviz" default="false"/>
  <arg name="enable_baxter" default="true"/>
  <arg name="limb" default="right"/>
    
  <!-- Enable baxter robot -->
  <group if="$(arg enable_baxter)">
    <node name="enable_baxter_robot" pkg="baxter_tools" type="enable_robot.py" output="screen" args="-e"/>
  </group>
  <group unless="$(arg enable_baxter)">
    <node name="enable_baxter_robot" pkg="baxter_tools" type="enable_robot.py" output="screen" args="-d"/>
  </group>
  
  <!-- Move the head -->
  <node name="command_head_pan" pkg="rostopic" type="rostopic" output="log"
        args="pub /robot/head/command_head_pan baxter_core_msgs/HeadPanCommand -- 0.0 10"/>
  
  <!-- PyKDL Jacobian Transpose Cartesian Controller -->
  <node name="jt_cartesian_controller" pkg="baxter_teleop" type="jt_cartesian_controller.py" output="screen">
    <rosparam command="load" file="$(find baxter_teleop)/config/jt_cartesian_controller.yaml"/>
    <param name="limb" type="string" value="$(arg limb)"/>
  </node>

  <!-- Show in Rviz -->
  <group if="$(arg rviz)">
    <node name="rviz" pkg="rviz" type="rviz" output="screen" args="-d $(find baxter_teleop)/config/cartesian_controller.rviz"/>
  </group>
</launch>
